### PR TITLE
chore: fix environment variable names for Sentry

### PR DESCRIPTION
**Pull request by starsirius**
The type of this PR is: **Chore**

This PR resolves [CX-692]

### Description

Discussion in [Slack](https://artsy.slack.com/archives/C01B2P6LJUU/p1602093820174400). The environment variable names for Sentry were incorrectly named as `SEGMENT_*`, instead of `SENTRY_*`.

### Post-merge migration
- [ ] Clean up deprecated `SEGMENT_PRODUCTION_DSN` and `SEGMENT_STAGING_DSN` in `.env.shared` and communicate to the team.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434

[CX-692]: https://artsyproduct.atlassian.net/browse/CX-692

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ Want to get the app running? Run this in your shell:
 git clone https://github.com/artsy/eigen.git
 cd eigen
 gem install bundler
-bundle install --without development distribution
+bundle install --without development
```
Comment (starsirius) [inline README.md:41]: Unrelated to this PR, but I think we can remove it because the distribution group was already removed from d45fdad.

**Comment by starsirius** [comment]
@ashfurrow and I paired on this, and we
- added the new variables `SENTRY_*_DSN` to CircleCI environment variables. The team will clean up deprecated ones in a separate effort.
- added the new variables to `.env.shared` on S3. The deprecated ones need to be removed _after_ this PR is merged to not break dev's local development.

File: CHANGELOG.yml [hunk 2]
```diff
@@ -2,34 +2,10 @@ upcoming:
...
-    - ConversationDetail component - sepand
-    - Adds "Artworks by Artists You Follow" rail (visible only to admins) - damon
-    - Upgrade Relay to 10 - chris
-    - Add share button to viewing rooms - mdole
-    - Add utms to share buttons - mdole
-    - Add new Show2 view, behind lab option - roop, sarah
-    - Adds tracking for Fair2 - will
-    - Adds ability to route to new fair view via option - sweir
-    - Include full artwork details in inquiry modal; improve animation - starsirius
-    - Adds route and view for all artworks by followed artists in a fair - will
```
Comment (starsirius) [inline CHANGELOG.yml:29]: I think these were [accidentally carried over](https://github.com/artsy/eigen/pull/4005/commits/5e67d70c5bdcc6e6109dd74a6d0a4a0148bb98eb) from 6.6.5. cc @sweir27 .

**Comment by starsirius** [comment]
OK this is ready for review again. Thanks for the patience and pairing, @ashfurrow! 🙏

File: CHANGELOG.yml (hunk 2 above)
Comment (sweir27) [reply CHANGELOG.yml:29]: 😱 very accidental! Unintentional rebase fallout. Ty @starsirius !

**Review by ashfurrow — APPROVED** [review]

**Pull request merged**